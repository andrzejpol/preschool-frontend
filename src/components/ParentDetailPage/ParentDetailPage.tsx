import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faCalendarDays, faDroplet, faEnvelope, faPhone, faPlus} from "@fortawesome/free-solid-svg-icons";

const ParentDetailPage = () => {

  return (
    <div className="flex-1 p-4 flex flex-col gap-4 xl:flex-row">
      <div className="w-full xl:2/3">
        <div className="flex flex-col lg:flex-row gap-4">
          <div className="py-6 px-4 rounded-md flex-1 bg-preschoolPrimaryLight flex gap-4">
            <div className="w-1/3">
              <img
                src="https://static.vecteezy.com/system/resources/previews/029/364/950/non_2x/3d-carton-of-boy-going-to-school-ai-photo.jpg"
                alt="" className="w-[144px] h-[144px] rounded-full object-cover"/>
            </div>
            <div className="w-2/3 flex flex-col justify-between gap-4">
              <h1 className="text-xl font-semibold">Leonard Snyder</h1>
              <p className="text-sm text-gray-500">Lorem ipsum, dolor sit amet. Concectetur adipiscing elit.</p>
              <div className="flex items-center justify-between gap-2 flex-wrap text-xs font-medium">
                <div className="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2">
                  <FontAwesomeIcon icon={faDroplet} className="flex items-center justify-center"/>
                  <span>A+</span>
                </div>
                <div className="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2">
                  <FontAwesomeIcon icon={faCalendarDays} className="flex items-center justify-center"/>
                  <span>January 2025</span>
                </div>
                <div className="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2">
                  <FontAwesomeIcon icon={faEnvelope} className="flex items-center justify-center"/>
                  <span>leonardsnyder@gmail.com</span>
                </div>
                <div className="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2">
                  <FontAwesomeIcon icon={faPhone} className="flex items-center justify-center"/>
                  <span>+48 609443221</span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="w-full xl:1/3 flex flex-col gap-4">
        <div className="bg-white p-4 rounded-md">
          <h1 className="text-xl font-semibold">Children</h1>
          <div className="mt-4 flex gap-4 flex-wrap text-xs text-gray-500">

          </div>
        </div>
      </div>
    </div>
  )
};

export default ParentDetailPage;