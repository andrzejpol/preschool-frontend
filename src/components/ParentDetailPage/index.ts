import ParentDetailPage from './ParentDetailPage';

export default ParentDetailPage;