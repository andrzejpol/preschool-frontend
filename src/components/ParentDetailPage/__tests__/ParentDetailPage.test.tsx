import React from "react";
import {render} from "@testing-library/react";
import ParentDetailPage from "../ParentDetailPage";

describe("ParentDetailPage Component", () => {
  it("matches the snapshot", () => {
    const {asFragment} = render(<ParentDetailPage/>);

    // Snapshot assertion
    expect(asFragment()).toMatchSnapshot();
  });
});
