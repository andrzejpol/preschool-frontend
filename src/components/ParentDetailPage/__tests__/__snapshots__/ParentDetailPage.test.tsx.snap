// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ParentDetailPage Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-1 p-4 flex flex-col gap-4 xl:flex-row"
  >
    <div
      class="w-full xl:2/3"
    >
      <div
        class="flex flex-col lg:flex-row gap-4"
      >
        <div
          class="py-6 px-4 rounded-md flex-1 bg-preschoolPrimaryLight flex gap-4"
        >
          <div
            class="w-1/3"
          >
            <img
              alt=""
              class="w-[144px] h-[144px] rounded-full object-cover"
              src="https://static.vecteezy.com/system/resources/previews/029/364/950/non_2x/3d-carton-of-boy-going-to-school-ai-photo.jpg"
            />
          </div>
          <div
            class="w-2/3 flex flex-col justify-between gap-4"
          >
            <h1
              class="text-xl font-semibold"
            >
              Leonard Snyder
            </h1>
            <p
              class="text-sm text-gray-500"
            >
              Lorem ipsum, dolor sit amet. Concectetur adipiscing elit.
            </p>
            <div
              class="flex items-center justify-between gap-2 flex-wrap text-xs font-medium"
            >
              <div
                class="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-droplet flex items-center justify-center"
                  data-icon="droplet"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 384 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M192 512C86 512 0 426 0 320C0 228.8 130.2 57.7 166.6 11.7C172.6 4.2 181.5 0 191.1 0l1.8 0c9.6 0 18.5 4.2 24.5 11.7C253.8 57.7 384 228.8 384 320c0 106-86 192-192 192zM96 336c0-8.8-7.2-16-16-16s-16 7.2-16 16c0 61.9 50.1 112 112 112c8.8 0 16-7.2 16-16s-7.2-16-16-16c-44.2 0-80-35.8-80-80z"
                    fill="currentColor"
                  />
                </svg>
                <span>
                  A+
                </span>
              </div>
              <div
                class="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-calendar-days flex items-center justify-center"
                  data-icon="calendar-days"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M128 0c17.7 0 32 14.3 32 32l0 32 128 0 0-32c0-17.7 14.3-32 32-32s32 14.3 32 32l0 32 48 0c26.5 0 48 21.5 48 48l0 48L0 160l0-48C0 85.5 21.5 64 48 64l48 0 0-32c0-17.7 14.3-32 32-32zM0 192l448 0 0 272c0 26.5-21.5 48-48 48L48 512c-26.5 0-48-21.5-48-48L0 192zm64 80l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm128 0l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0zM64 400l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0zm112 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16z"
                    fill="currentColor"
                  />
                </svg>
                <span>
                  January 2025
                </span>
              </div>
              <div
                class="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope flex items-center justify-center"
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48L48 64zM0 176L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-208L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                    fill="currentColor"
                  />
                </svg>
                <span>
                  leonardsnyder@gmail.com
                </span>
              </div>
              <div
                class="w-full md:w-1/3 lg:w-full 2xl:w-1/3 flex items-center gap-2"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-phone flex items-center justify-center"
                  data-icon="phone"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M164.9 24.6c-7.7-18.6-28-28.5-47.4-23.2l-88 24C12.1 30.2 0 46 0 64C0 311.4 200.6 512 448 512c18 0 33.8-12.1 38.6-29.5l24-88c5.3-19.4-4.6-39.7-23.2-47.4l-96-40c-16.3-6.8-35.2-2.1-46.3 11.6L304.7 368C234.3 334.7 177.3 277.7 144 207.3L193.3 167c13.7-11.2 18.4-30 11.6-46.3l-40-96z"
                    fill="currentColor"
                  />
                </svg>
                <span>
                  +48 609443221
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full xl:1/3 flex flex-col gap-4"
    >
      <div
        class="bg-white p-4 rounded-md"
      >
        <h1
          class="text-xl font-semibold"
        >
          Children
        </h1>
        <div
          class="mt-4 flex gap-4 flex-wrap text-xs text-gray-500"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
